kratos/geometries/quadrilateral_2d_4.h

```diff
@@ -511,6 +512,32 @@ template<class TPointType> class Quadrilateral2D4
         return edges;
     }
 
+    /** This method checks if an axis-aliged bounding box (AABB)
+    intersects the quadrilateral
+
+    @return bool if the quadrilateral overlaps the box
+    @param rLowPoint first corner of the box
+    @param rHighPoint second corner of the box
+    @see Triangle2D3::HasIntersection
+    */
+    bool HasIntersection( const Point& rLowPoint, const Point& rHighPoint ) override
+    {
+        typename Triangle2D3<PointType>::Pointer p_triangle_0 (new Triangle2D3<PointType>(
```

Comment:
It would be much faster to use an stack variable instead of new allocated object in heap:

```c++
Triangle2D3<PointType> triangle_0(this->pGetPoint( 0 ),
                                  this->pGetPoint( 1 ),
                                  this->pGetPoint( 2 ));

if( triangle_0.HasIntersection(rLowPoint, rHighPoint) ) 
    return true;
```

Please consider that memory allocation could take much longer time than the entire arithmetic you have in this function.

Reply:
thanks, I have modified it